thor/main.py

```diff
@@ -137,42 +137,121 @@ def clusterVelocity(
         If clusters are found, will return a list of numpy arrays containing the 
         observation IDs for each cluster. If no clusters are found, will return np.NaN.
     """ 
+    logger.debug(f"cluster: vx={vx} vy={vy} n_obs={len(obs_ids)}")
     xx = x - vx * dt
     yy = y - vy * dt
+
+    X = np.stack((xx, yy), 1)
+
+    clusters = find_clusters(X, eps, min_samples, alg=alg)
+    clusters = filter_clusters_by_length(clusters, dt, min_samples, min_arc_length)
+
+    cluster_ids = []
+    for cluster in clusters:
+        cluster_ids.append(obs_ids[cluster])
+
+    if len(cluster_ids) == 0:
+        cluster_ids = np.NaN
```

Comment:
I left this as is, but it doesn't really make sense to me. An empty list would seem more right?

Reply:
It's super counterintuitive that's for sure. I'm returning a `NaN` so I can use `pandas.Series.isna()` to filter out velocity combinations that returned no clusters (this is done in that mess of code that restructures the clusters). I did not put much effort into finding an equivalent that could filter out an empty list. 

There is a broader point I think worth mentioning: I'm not a fan of my restructuring code since it's hard to understand. As THOR matured I have settled on a two dataframe structure for linkages: linkages and linkage_members. The restructuring code transforms a list of lists and nans into those two dataframes. Depending on how much of a speed-up you are able to find while clustering, we might want to consider moving that dataframe construction into the clusterVelocity function or into the worker functions so we can avoid the messy wrangling.